### PR TITLE
Feature/add post

### PR DESCRIPTION
**Pull request by owenhallgren**
## Purpose
This feature is adding forms for the user to be able post their own data in the back-end server

## Abstract
This adds three forms for the user to input a date they want to add the data too, and submit other needed information. The forms will make sure all the right data is needed before submitting the post. 

#### Pre-Merge TODOs
- [ ] Think about best ways to make this look good and match the current UI
- [ ] Check Styling/Syntax
- [ ] Check for opportunities for further refactoring

## Learning
Most time was spent researching the functionality of POST, and how to integrate it with user inputs 

#### Next Steps
It does not look good at all, needs complete styling and could use a more in-depth error catch method

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -150,13 +151,88 @@ let stepsTrendingButton = document.querySelector('.steps-trending-button');
...
 profileButton.addEventListener('click', showDropdown);
 stairsTrendingButton.addEventListener('click', updateTrendingStairsDays());
 stepsTrendingButton.addEventListener('click', updateTrendingStepDays());
 
+submitButton.addEventListener('click', postSleepHelper );
+hydrationSubmitButton.addEventListener('click', postHydrationHelper);
+activitySubmitButton.addEventListener('click', postActivityHelper);
+showSleepFormButton.addEventListener('click', showSleepForm);
+showActivityFormButton.addEventListener('click', showActivityForm);
+showHydrationFormButton.addEventListener('click', showHydrationForm);
```
Comment (elizhahn) [inline src/scripts.js:188]: Could possibly use event delegation for this??

File: src/scripts.js [hunk 2]
```diff
@@ -150,13 +151,88 @@ let stepsTrendingButton = document.querySelector('.steps-trending-button');
...
+}
+
+function showHydrationForm() {
+  hydrationInputForm.classList.remove("hide");
+  activityInputForm.classList.add("hide");
+  sleepInputForm.classList.add("hide");
+  successfulSubmit.classList.add("hide");
+  failedSubmit.classList.add("hide");
+  clearButton.classList.remove("hide");
+  errorMessage.classList.add("hide")
```
Comment (elizhahn) [inline src/scripts.js:232]: probably dry this up too and create a function maybe that can hide

**Review by elizhahn — APPROVED** [review]

**Review by ConnorAndersonLarson — APPROVED** [review]
Looks good from what I've seen

File: src/scripts.js (hunk 1 above)
Comment (owenhallgren) [reply src/scripts.js:188]: Defiantly, I was thinking we do this in the script refactor.

File: src/scripts.js (hunk 2 above)
Comment (owenhallgren) [reply src/scripts.js:232]: Script refactor!

**Pull request merged**